src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerProcessor.java

```diff
@@ -58,35 +61,79 @@ public void process(String containerName) {
+        Instant blobCreationDate = blobClient.getProperties().getLastModified().toInstant();
         if (blobReadinessChecker.isReady(blobCreationDate)) {
             return true;
         } else {
             logger.info(
                 "Blob not ready to be processed yet, skipping. File name: {}. Container: {}",
-                blob.getName(),
+                blobClient.getName(),
                 containerName
             );
             return false;
         }
     }
 
-    private void processBlob(BlobClient blob) {
-        envelopeService
-            .findLastEnvelope(blob.getBlobName(), blob.getContainerName())
+    private void processBlob(BlobClient blobClient) {
+        getLastEnvelope(blobClient)
             .ifPresentOrElse(
                 envelope -> {
                     if (envelope.status == Status.CREATED) {
-                        leaseAndThen(blob, () -> blobProcessor.continueProcessing(envelope.id, blob));
+                        leaseAndThen(blobClient, () ->
+                            continueProcessingEnvelopeIfEligible(
+                                blobClient,
+                                this::logEnvelopeDeleted
+                            )
+                        );
                     } else {
-                        logger.info("Envelope already processed in system, skipping. {} ", envelope.getBasicInfo());
+                        logEnvelopeProcessed(envelope);
                     }
                 },
-                () -> leaseAndThen(blob, () -> blobProcessor.process(blob))
+                () -> leaseAndThen(blobClient, () ->
+                    continueProcessingEnvelopeIfEligible(
+                        blobClient,
+                        blobProcessor::process
+                    )
+                )
             );
     }
 
+    private void continueProcessingEnvelopeIfEligible(
+        BlobClient blobClient,
+        Consumer<BlobClient> nonExistingEnvelopeHandler
+    ) {
+        getLastEnvelope(blobClient)
+            .ifPresentOrElse(
+                envelopePotentiallyProcessed -> continueProcessingIfPossible(blobClient, envelopePotentiallyProcessed),
+                () -> nonExistingEnvelopeHandler.accept(blobClient)
+            );
+    }
+
+    private void continueProcessingIfPossible(BlobClient blobClient, Envelope envelope) {
+        if (envelope.status == Status.CREATED) {
+            blobProcessor.continueProcessing(envelope.id, blobClient);
+        } else {
+            logEnvelopeProcessed(envelope);
+        }
+    }
+
+    private void logEnvelopeDeleted(BlobClient blobClient) {
+        logger.error(
+            "Envelope deleted in system for blob {}, container {}",
+            blobClient.getBlobName(), blobClient.getContainerName()
+        );
+    }
+
+    private void logEnvelopeProcessed(Envelope envelope) {
+        logger.info("Envelope processed in system, skipping. {} ", envelope.getBasicInfo());
```

Comment:
`already` was actually good here and in the methods's name - it adds useful information. It was just misleading for deletion, as deletion should never happen.